docs/kbs_attestation_protocol.md

```diff
@@ -1,220 +1,402 @@
+The [authentication](#authentication) phase is the first step of the overall KBS
+protocol, where attesters authenticate themselves against a KBS implementation.
+This leads to the second phase, during which authenticated attesters can request
+protected resources from the KBS.
+
+The KBS implementation keeps track of attestation results and binds them to a
+cookie identifier. During the second phase, the KBS can then decide if a
+specific resource could be released to a given attester, by mapping the cookie
+identifier the attester includes in its resource request message to its
+attestation results.
+
+To request a protected resource from the KBS, the attester sends a `GET` request
+to a resource specific endpoint. If the attester is allowed to access the
+resource, the KBS will respond to the `GET` request with an HTTP response which
+content is set to a [KBS Response](#response) JSON payload:
+
+```
+┌──────────┐                                         ┌─────┐
+│ Attester │                                         │ KBS │
+└───┳──────┘                                         └──┳──┘
+    ┃            ┌─────────────────────────┐            ┃
+    ┃────────────│GET /kbs/v0/<resource_id>│───────────▶┃
+    ┃      ┌─────┴─────────────────────────┴────┐       ┃
+    ┃      │Header: "Cookie kbs-session-id=123" │       ┃
+    ┃      │Body: N/A                           │       ┃
+    ┃      └────────────────────────────────────┘       ┃
+    ┃                    ┌────────┐                     ┃
+    ┃◀───────────────────│Response│─────────────────────┃
+    ┃     ┌──────────────└────────┘───────────────┐     ┃
+    ┃     │Header: "Set-Cookie kbs-session-id=123"│     ┃
+    ┃     │Body: Response                         │     ┃
+    ┃     └───────────────────────────────────────┘     ┃
+    ┃                                                   ┃
+    ┃                                                   ┃
+    ┃                                                   ┃
+    ▽        Figure 2: KBS Resource Request Phase       ▽
+```
+
+A request for protected resource can fail for three reasons:
+
+1. The requester is not authenticated and thus can not provide the right cookie
+   identifier. The KBS implementation sends an HTTP response with a 401
+   (`Unauthorized`) status code.
+2. The attester is authenticated but requests a resource that it's not allowed
+   to receive. The KBS implementation sends an HTTP response with a 403
+   (`Forbidden`) status code.
+3. The requested resource does not exist. The KBS implementation sends an HTTP
+   response with a 404 (`Not Found`) status code.
+
+The KBS protocol currently supports two kinds of resources for an attester to
+request: keys and tokens.
+
+### Key Resource
+
+Keys are generic protected resources that an authenticated attester can get by
+sending a `GET` HTTP request to the `/kbs/v0/key/<key_id>` endpoint.
+
+The `<key_id>` resource name is defined by the attestation service. There should
```

Comment:
There may be some confusion here. The attestation service should only be responsible for the verification of the evidence in the `Attestation` step, and the resource name should be defined by KBS itself.

Reply:
Thanks, I'll fix this as this could be quite confusing.

Reply:
Fixed, thanks.